src/Parsers/ExpressionListParsers.cpp

```diff
@@ -1046,6 +1061,24 @@ class FunctionLayer : public Layer
 
     bool allow_function_parameters;
     bool is_compound_name;
+
+    enum NullsAction
+    {
+        DEFAULT = 0,
+        RESPECT_NULLS = 1,
+        IGNORE_NULLS = 2,
+    };
+    static String transformFunctionNameForRepectNulls(const String & original_function_name, NullsAction nulls_action)
+    {
+        static std::unordered_map<String, std::vector<String>> renamed_functions_with_nulls = {
+            {"first_value", {"first_value_ignore_nulls", "first_value_respect_nulls", "first_value_ignore_nulls"}},
+            {"last_value", {"last_value_ignore_nulls", "last_value_respect_nulls", "last_value_ignore_nulls"}},
```

Comment:
Can we leave current function name with no modifier? To have the same column name as before `first_value(x)` in such case

```suggestion
            {"first_value", {"first_value", "first_value_respect_nulls", "first_value_ignore_nulls"}},
            {"last_value", {"last_value", "last_value_respect_nulls", "last_value_ignore_nulls"}},
```

Reply:
The function factor use the first element to look up the function, it conflicts with previous `first_value`